client/client_test.go

```diff
@@ -4860,6 +4862,44 @@ func testRelativeMountpoint(t *testing.T, sb integration.Sandbox) {
 	require.Equal(t, dt, []byte(id))
 }
 
+func testExporterImageDescriptor(t *testing.T, sb integration.Sandbox) {
+	skipDockerd(t, sb)
+	requiresLinux(t)
+	c, err := New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	st := llb.Image("busybox:latest")
+	def, err := st.Marshal(sb.Context())
+	require.NoError(t, err)
+
+	res, err := c.Solve(sb.Context(), def, SolveOpt{
+		Exports: []ExportEntry{
+			{
+				Type:  ExporterOCI,
+				Attrs: map[string]string{},
+				Output: func(m map[string]string) (io.WriteCloser, error) {
+					return nil, nil
+				},
+			},
+		},
+	}, nil)
+	require.NoError(t, err)
+
+	require.Contains(t, res.ExporterResponse, exptypes.ExporterImageDescriptorKey)
+	dt, err := base64.StdEncoding.DecodeString(res.ExporterResponse[exptypes.ExporterImageDescriptorKey])
```

Comment:
There are integration tests using `buildctl` as well https://github.com/moby/buildkit/blob/master/cmd/buildctl/build_test.go#L48 that allow you to also cover the json decoding feature without reimplementing it in a test.

Reply:
I've already expanded the `testBuildMetadataFile` one for the descriptor: https://github.com/moby/buildkit/pull/2610/files#diff-487f6d73514b6a9c5895f8c77586ce38dc2b108fb3ed93bd0d63452667a7c3c3R157-R164. Is it not enough?

Reply:
No that is fine. I'm ok with just removing this test completely. All cases seem to be covered also in that buildctl test.